src/core/instance/state.js

```diff
@@ -143,12 +145,17 @@ function initMethods (vm: Component) {
   if (methods) {
     for (const key in methods) {
       vm[key] = methods[key] == null ? noop : bind(methods[key], vm)
-      if (process.env.NODE_ENV !== 'production' && methods[key] == null) {
-        warn(
+      if (process.env.NODE_ENV !== 'production') {
+        methods[key] == null && warn(
           `method "${key}" has an undefined value in the component definition. ` +
           `Did you reference the function correctly?`,
           vm
         )
+        hasOwn(BuiltinVue.prototype, key) && warn(
+          `You're overriding Vue's internal method "${key}". ` +
+          `Beware of misbehaviors.`,
```

Comment:
Let rephrase this to "Avoid overriding Vue's internaml method ..." and can remove the second line.